### PR TITLE
#3924 - Ecert Creation - "Prefer not to answer" Gender

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Update the gender to have a white space as a default option.
- Adjusted and created some E2E tests for e-Certs.
- Adjusted MSFAA E2E that was already using "preferNotToAnswer".
- SIN is using the same method from e-Cert and MSFAA.

### CODE REVIEW

File: sources/packages/backend/libs/utilities/src/esdc-utils.ts [hunk 1]
```diff
@@ -32,7 +32,10 @@ export function getGenderCode(gender: string): string {
     case "nonBinary":
       return "X";
     default:
-      return "";
+      // Gender should have at least one character code.
+      // The default option as an empty space would represent
+      // the "Prefer not to answer"(preferNotToAnswer) option.
+      return " ";
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/utilities/src/esdc-utils.ts:38]: 👍

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts [hunk 2]
```diff
@@ -410,6 +411,70 @@ describe(
...
+      // Assert
+      // Assert student profile data.
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      expect(uploadedFile.fileLines).toHaveLength(3);
+      const [, record] = uploadedFile.fileLines;
+      // TODO: include other student profile fields as needed.
+      const recordParsed = new PartTimeCertRecordParser(record);
+      expect(recordParsed.recordType).toBe("02");
+      expect(recordParsed.hasUser(student.user)).toBe(true);
+      expect(recordParsed.gender).toBe(" ");
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts:475]: 👍

**Review by dheepak-aot — APPROVED** [review]
Thanks for adding the E-cert E2E test. Looks good. 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Pull request merged**